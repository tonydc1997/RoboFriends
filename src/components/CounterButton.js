import React, { Component } from 'react';

class CounterButton extends Component {
  constructor() {
    super();
    this.state = {
      count: 1
    }
  }

  shouldComponentUpdate(nextProps, nextState) {
    return true;
  }

  updateCount = () => {
    this.setState({count: this.state.count + 1})
  }

  render() {
    console.log('CounterButton');
    return (
      <button className='br2 bg-light-green b--navy pb1 pt1 grow' onClick={this.updateCount} >
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;Â 
