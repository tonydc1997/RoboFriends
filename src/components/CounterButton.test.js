import React from 'react';
import { shallow } from 'enzyme';
import CounterButton from './CounterButton';

describe('CounterButton component', () => {
  it('Counter: renders correctly', () => {
    expect(shallow(<CounterButton />)).toMatchSnapshot();
  })
})