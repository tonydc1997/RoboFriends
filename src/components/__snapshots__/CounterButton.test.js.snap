// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component Counter: renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Count: ",
        1,
      ],
      "className": "br2 bg-light-green b--navy pb1 pt1 grow",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Count: ",
      1,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Count: ",
          1,
        ],
        "className": "br2 bg-light-green b--navy pb1 pt1 grow",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Count: ",
        1,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
